BookReader/BookReader.js

```diff
@@ -2948,12 +2952,23 @@ BookReader.prototype.search = function(term, options) {
     term = term.replace(/['"]+/g, '');
     term = '"' + term + '"';
 
-    var url = 'https://'+this.server.replace(/:.+/, ''); //remove the port and userdir
-    //url    += '/fulltext/inside.php?item_id='+this.bookId;
-    url    += '/fulltext/new_inside.php?item_id='+this.bookId;
-    url    += '&doc='+this.subPrefix;   //TODO: test with subitem
-    url    += '&path='+this.bookPath.replace(new RegExp('/'+this.subPrefix+'$'), ''); //remove subPrefix from end of path
-    url    += '&q='+escape(term);
+    // Remove the port and userdir
+    var url = 'https://' + this.server.replace(/:.+/, '') + this.searchInsideUrl + '?';
+
+    // Remove subPrefix from end of path
+    var path = this.bookPath;
+    if (this.bookPath.length - this.bookPath.lastIndexOf(this.subPrefix) == this.subPrefix.length) {
+      path = this.bookPath.substr(0, this.bookPath.length - this.subPrefix.length);
+    }
+
+    var urlParams = {
```

Comment:
Suggestion: Might want to move the `var` up to avoid any hoisting problems.

Reply:
I'm not sure I understand. Wouldn't just using `urlParams` without `var` be syntactically incorrect?

Reply:
You’re familiar with variable hoisting in JavaScript, right?

Basically, right now `urlParams` is actually being declared at the top of your function scope, and on this line you’re only assigning its value, due to how JS works. I’m suggesting the common JS best practice of moving the var declaration (not necessarily the value assignment) to the top of the function, or at least closer to the top, to prevent any bugs related to this.

Reply:
Yes I'm familiar with hoisting (you actually explained it to me previously). I don't think it will cause an issue here, and most of the rest of the code is not accounting for it either, so I think we should leave it as is.

Reply:
I also misread that earlier. I thought you said "might want to *remove the var*". heh

Reply:
Ah, gotcha. To make my general position clear: Vars should ALWAYS be at the top, not only when context demands it. Not saying that has to happen here though. I understand not wanting to move too many things around.